pkg/manager/manager_test.go

```diff
@@ -214,197 +324,194 @@ func Test_LoadManagerBadNetworkStore(t *testing.T) {
-	assert.Equal(t, change1.Description, "Originally created as part of Test_SetNetworkConfig")
+	// Verify the change
+	validationError := mgrTest.ValidateNewNetworkConfig(Device1, DeviceVersion, DeviceType, updates, deletes)
+	assert.NilError(t, validationError, "ValidateTargetConfig error")
 
-	assert.Equal(t, string(*configName), "Device1-1.0.0")
+	// Set the new change
+	const testNetworkChange network.ID = "Test_SetNetworkConfig"
+	err := mgrTest.SetNewNetworkConfig(updatesForDevice1, deletesForDevice1, deviceInfo, string(testNetworkChange))
+	assert.NilError(t, err, "SetTargetConfig error")
+	testUpdate, _ := mgrTest.NetworkChangesStore.Get(testNetworkChange)
+	assert.Assert(t, testUpdate != nil)
+	assert.Equal(t, testUpdate.ID, testNetworkChange, "Change Ids should correspond")
 
-	//Done in this order because ordering on a path base in the store.
-	updatedVals := changeStoreTest[store.B64(changeID)].Config
+	// Check that the created change is correct
+	updatedVals := testUpdate.Changes[0].Values
 	assert.Equal(t, len(updatedVals), 2)
 
-	//Asserting change
+	// Asserting update to 2A
 	assert.Equal(t, updatedVals[0].Path, Test1Cont1ACont2ALeaf2A)
 	assert.Equal(t, (*devicechangetypes.TypedFloat)(updatedVals[0].GetValue()).Float32(), float32(ValueLeaf2B314))
 	assert.Equal(t, updatedVals[0].Removed, false)
 
-	// Checking original is still alright
-	originalChange, ok = changeStoreTest[origChangeHash]
-	assert.Assert(t, ok)
-	assert.Equal(t, (*devicechangetypes.TypedFloat)(originalChange.Config[2].GetValue()).Float32(), float32(ValueLeaf2B159))
-
-	//Asserting deletion of 2C
+	// Asserting deletion of 2C
 	assert.Equal(t, updatedVals[1].Path, Test1Cont1ACont2ALeaf2C)
 	assert.Equal(t, (*devicechangetypes.TypedEmpty)(updatedVals[1].GetValue()).String(), "")
 	assert.Equal(t, updatedVals[1].Removed, true)
-
 }
 
 // When device type is given it is like extension 102 and allows a never heard of before config to be created
 func Test_SetNetworkConfig_NewConfig(t *testing.T) {
-	mgrTest, changeStoreTest, configurationStoreTest, _, _, _ := setUp(t)
+	mgrTest, _, _, _ := setUp(t)
+
+	// Making change
+
+	const Device5 = "Device5"
+	const NetworkChangeAddDevice5 = "NetworkChangeAddDevice5"
 
 	updates := make(devicechangetypes.TypedValueMap)
-	deletes := make([]string, 0)
+	updates[Test1Cont1ACont2ALeaf2A] = devicechangetypes.NewTypedValueFloat(ValueLeaf2B314)
+	deletes := []string{Test1Cont1ACont2ALeaf2C}
 
-	// Making change
-	updates[Test1Cont1ACont2ALeaf2A] = (*devicechangetypes.TypedValue)(devicechangetypes.NewTypedValueFloat(ValueLeaf2B314))
-	deletes = append(deletes, Test1Cont1ACont2ALeaf2C)
+	updatesForDevice, deletesForDevice, deviceInfo := makeDeviceChanges(Device5, updates, deletes)
 
-	changeID, configName, err := mgrTest.SetNetworkConfig("Device5", "1.0.0", "Devicesim", updates, deletes, "Test_SetNetworkConfig_NewConfig")
+	err := mgrTest.SetNewNetworkConfig(updatesForDevice, deletesForDevice, deviceInfo, NetworkChangeAddDevice5)
```

Comment:
`deviceInfo` is nil here - it should be "1.0.0" and "TestDevice"
in fact there is an error in Manager.computeNewNetworkConfig() that is ignoring it, and also it's putting the delete in to the update change - i'll tackle this in a separate PR